### PR TITLE
fix(server): detection new MS Edge Chromium

### PR DESCRIPTION
**Pull request by anthony-redFox**
BREAKING CHANGES - Browser and OS name generation are changed
swich on new lib ua-parser-js to have a good detection of MS Edge Chromium

### CODE REVIEW

**Comment by anthony-redFox** [comment]
@johnjbarton  please take a look on our(and @devoto13) PRs

File: lib/helper.js [hunk 1]
```diff
@@ -3,14 +3,16 @@
...
-const useragent = require('useragent')
+const useragent = require('ua-parser-js')
 const Promise = require('bluebird')
 const mm = require('minimatch')
 
 exports.browserFullNameToShort = (fullName) => {
-  const agent = useragent.parse(fullName)
-  const isUnknown = agent.family === 'Other' || agent.os.family === 'Other'
-  return isUnknown ? fullName : `${agent.toAgent()} (${agent.os})`
+  const au = useragent(fullName)
```
Comment (abaksha-sc) [inline lib/helper.js]: Not critical, but did you mean `ua` instead of `au`?

File: test/e2e/upstream-proxy.feature [hunk 2]
```diff
@@ -7,7 +7,7 @@ Feature: UpstreamProxy
     Given a configuration with:
       """
       files = ['behind-proxy/plus.js', 'behind-proxy/test.js'];
-      browsers = ['ChromeHeadlessNoSandbox'];
+      browsers = ['ChromeHeadless'];
```
Comment (johnjbarton) [inline test/e2e/upstream-proxy.feature]: Is this commit related to the previous one? Why this change?

Comment (anthony-redFox) [reply test/e2e/upstream-proxy.feature]: I tried to fix the issue with "Error: socket hang up" in e2e test
Builds like 
https://travis-ci.org/github/karma-runner/karma/jobs/650593759?utm_medium=notification&utm_source=github_status
https://travis-ci.org/github/karma-runner/karma/jobs/659054766?utm_medium=notification&utm_source=github_status

I am ok, if need to remove last commit. Do I need to do it? @johnjbarton

Comment (johnjbarton) [reply test/e2e/upstream-proxy.feature]: Please create a PR for this change and remove this last commit. I will review and land the one related to e2e first.
